src/main/java/com/wavefront/sdk/jersey/WavefrontJerseyFilter.java

```diff
@@ -38,13 +56,41 @@
   private final ApplicationTags applicationTags;
   private final ThreadLocal<Long> startTime = new ThreadLocal<>();
   private final ThreadLocal<Long> startTimeCpuNanos = new ThreadLocal<>();
+  private final ThreadLocal<Boolean> finished = ThreadLocal.withInitial(() -> false);
```

Comment:
I don't see the need for this additional protection as finishing already finished span should be Noop. If you feel this is necessary store the state along with Span in SpanWrapper when you store it in requestContext instead of storing the state in two places.